### PR TITLE
feat: enable separate configuration of Snyk Code Quality and Snyk Code Security issues and filter reporting

### PR DESCRIPTION
**Pull request by bastiandoetsch**
### Description

_Provide description of this PR and changes, if linked Jira ticket doesn't cover it in full._

### Checklist

- [x] Tests added and all succeed
- [x] Linted
- [x] README.md updated, if user-facing
- [x] License file updated, if new 3rd-party dependency is introduced

### Screenshots / GIFs

_Visuals that may help the reviewer. Please add screenshots for any UI change. GIFs are most welcome!_

### CODE REVIEW

File: application/config/config.go [hunk 1]
```diff
@@ -239,7 +241,9 @@ func (c *Config) loadFile(fileName string) {
 		log.Info().Str("method", "loadFile").Msg("Couldn't load " + fileName)
 		return
 	}
-	defer file.Close()
+	defer func(file *os.File) {
+		_ = file.Close()
+	}(file)
```
Comment (asaf92) [inline application/config/config.go:246]: Why? The error is still discarded

File: infrastructure/code/convert.go [hunk 2]
```diff
@@ -278,16 +279,25 @@ func (s *SarifResponse) toIssues() (issues []snyk.Issue) {
 				Severity:            issueSeverity(result.Level),
 				Message:             message,
 				FormattedMessage:    formattedMessage,
-				IssueType:           snyk.CodeSecurityVulnerability,
+				IssueType:           snyk.CodeSecurityVulnerability, // FIXME: This is incorrect, we need to differentiate (see https://snyksec.atlassian.net/browse/ROAD-1161)
```
Comment (asaf92) [inline infrastructure/code/convert.go:282]: This needs to be implemented

File: application/config/config.go (hunk 1 above)
Comment (bastiandoetsch) [reply application/config/config.go:250]: Because I get warnings in goland :)

File: infrastructure/code/convert.go (hunk 2 above)
Comment (bastiandoetsch) [reply infrastructure/code/convert.go:282]: Yes, but it's a different story

File: application/config/config.go (hunk 1 above)
Comment (asaf92) [reply application/config/config.go:250]: If Goland was any good you'd still get them 🤣

Comment (asaf92) [reply application/config/config.go:250]: (jk I love Goland)

**Pull request merged**